### PR TITLE
🚀 Create definition element for pool-bound <amp-video>

### PR DESCRIPTION
**Pull request by alanorozco**
The current flow for media pool registry + insertion is as follows

This is problematic since the first step unnecessarily adds a media element to the DOM. Since the media pool limits the amount of media element per document, in cases where the media component count is greater than the media allowed by the media pool there will be performance issues.

1. A media element is created by the AMP media component. (**Performance hit**)
2. The media element is detached and kept as a reference element. (**Performance hit**)
3. A different media element is inserted, whose properties and children are copies of the reference element. (Real playable element is attached.)

This PR simplifies this flow and improves performance (steps 1 and 2) by directly providing reference elements to media pool that are detached from the DOM. 

**Work in progress.** Depends on mediapool understanding the concepts of media definition references and act accordingly (e.g. not detaching, keeping reference of parent `amp-video`, etc).

### CODE REVIEW

File: extensions/amp-video/0.1/amp-video.js [hunk 1]
```diff
@@ -188,7 +188,9 @@ class AmpVideo extends AMP.BaseElement {
 
     this.configure_();
 
-    this.video_ = element.ownerDocument.createElement('video');
+    const video = this.createVideoOrDefinitionElement_();
+
+    this.video_ = video;
```
Comment (newmuis) [inline extensions/amp-video/0.1/amp-video.js]: Just curious: why not just use `this.video_` everywhere?

Comment (alanorozco) [reply extensions/amp-video/0.1/amp-video.js]: Minified bundle size :) By using a scoped reference we leverage the minified reference name (which should be one or two chars) vs `_this3.video`.

It shouldn't make too much of a difference with this method alone but went ahead and made the change for consistency.

**Comment by alanorozco** [comment]
@mrjoro This is a WIP blocked on work from the `amp-story` side. I'll be responsible for keeping this PR evergreen.

**Comment by nainar** [comment]
Closing this PR as inactive. Please reopen if needed.

**Pull request closed without merging**